services/src/main/java/org/keycloak/compatibility/KeycloakCompatibilityMetadataProvider.java

```diff
@@ -1,22 +1,70 @@
 package org.keycloak.compatibility;
 
 import java.util.Map;
+
+import org.keycloak.common.Profile;
 import org.keycloak.common.Version;
+import org.keycloak.migration.ModelVersion;
+import org.keycloak.migration.ModelVersionUtils;
 
 /**
  * A {@link CompatibilityMetadataProvider} implementation to provide the Keycloak version.
  */
 public class KeycloakCompatibilityMetadataProvider implements CompatibilityMetadataProvider {
 
     public static final String ID = "keycloak";
+    public static final String VERSION_KEY = "version";
+    private final String version;
+
+    public KeycloakCompatibilityMetadataProvider() {
+        this(Version.VERSION);
+    }
+
+    public KeycloakCompatibilityMetadataProvider(String version) {
+        this.version = version;
+    }
 
     @Override
     public Map<String, String> metadata() {
-        return Map.of("version", Version.VERSION);
+        return Map.of(VERSION_KEY, version);
+    }
+
+    @Override
+    public CompatibilityResult isCompatible(Map<String, String> other) {
+        CompatibilityResult equalComparison = CompatibilityMetadataProvider.super.isCompatible(other);
+
+        // If V2 feature is enabled, we consider versions upgradable in rolling way also if other is the previous micro release
+        if (Profile.isFeatureEnabled(Profile.Feature.ROLLING_UPDATES_V2) && Util.isNotCompatible(equalComparison)) {
```

Comment:
This is not very future-proof. If we add more information to this class, the compatibility may fail because of something else and not the version; it will return the wrong outcome. 

The question that follows is... are we planning to add more metadata to this class?

Reply:
I added a check for it. I think we can easily miss this when doing some addition the `KeycloakCompatibilityProvider` later.